06-security.md

```diff
@@ -1 +1,451 @@
+
+### Check Object Access Permissions 
+
+In a typical CRUD system, access is granted on a collection (or type) basis. So if a user can edit one object of a type, its possible to edit them all. 
+This coarse grained access mechanism is not suitable if a resource is owned by a specific user. If this is the case additional checks must be made at the service layer.
+
+### Validate Strings 
+
+An innocent looking unvalidated String field such as 'name' may contain any character, including newline's null char's and exotic symbols. These characters may have interesting effects in downstream (legacy) systems or libraries. (such as [JavaMail Header Injection via Subject](http://www.csnc.ch/misc/files/advisories/CSNC-2014-001_javamail_advisory_public.txt)). In general, only allow characters that have meaning for the field and reject any unwanted ones (especially those below ASCII value 32) and limit the length of the field. Prefer whitelisting allowable characters above blacklisting illegal ones.   
+
+### Filter HTML 
+
+If the application allows rich HTML editing make sure that the server side sanitizes the HTML. Various libraries exist to do this, for example [JSoup](http://jsoup.org/cookbook/cleaning-html/whitelist-sanitizer). If you just want to block unsafe HTML you can use the `SafeHtml` annotation from hibernate-validator.
+
+# Implementing the Guidelines with Spring Security
+
+Spring Security 4 has a lot of sensible defaults which make implementing these guidelines relatively easy. Accompanied with this chapter is a working example web application which implements the recommendations.    
+
+## Use HTTPS
+
+In order to use HTTPS you need to configure your web container with a certificate. 
+When developing a self-signed certificate is sufficient, for production purposes a certificate signed by a certificate authority is required.
+
+The Tomcat7 website has a good [how-to](https://tomcat.apache.org/tomcat-7.0-doc/ssl-howto.html) for a self-signed certificate and also detailed instructions for a production certificate.
+
+Creating a self-signed certificate in a keystore for development can be done using the Java `keystore` command; enter the following and follow instructions.
+```bash
+keytool -genkey -alias tomcat -keyalg RSA -keystore keystore.jks
+```
+It will provide you with a `keystore.jks` containing the certificate. 
+
+You will need it if you want to configure your development environment:
+* The [tomcat7-maven-plugin](https://tomcat.apache.org/maven-plugin-2.0/tomcat7-maven-plugin/run-mojo.html) is configured using the httpsPort, keystoreFile and keystorePass properties. 
+* Spring-Boot allows [configuration via properties](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-embedded-servlet-containers.html#howto-configure-ssl).  
+
+You can set Spring security to accept only HTTPS traffic with the following configuration snippet: 
```

Comment:
Is the link in the 'configure your development environment'  section not clear enough?

Also, the sample project has a working configuration.